apps/console/src/features/identity-providers/components/forms/helpers/form-fields-helper.tsx

```diff
@@ -488,8 +521,89 @@ export const getDropDownField = (eachProp: CommonPluggableComponentPropertyInter
                 disabled={ propertyMetadata?.isDisabled }
                 readOnly={ propertyMetadata?.readOnly }
                 data-testid={ `${ testId }-${ propertyMetadata?.key }` }
-            />
-            { propertyMetadata?.description && (
+            />) }
+            {
+                showField
+                && propertyMetadata?.description
+                && (
+                    <Hint disabled={ propertyMetadata?.isDisabled }>
+                        { propertyMetadata?.description }
+                    </Hint>
+                )
+            }
+        </>
+    );
+};
+
+export const getDropDownFieldWithListener = (eachProp: CommonPluggableComponentPropertyInterface,
+    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
+    listen: (key: string, values: Map<string, FormValue>) => void,
+    testId?: string,
+    showField?: boolean): ReactElement => {
+    return (
+        <>
+            { showField && (<Field
+                name={ eachProp?.key || propertyMetadata?.key }
+                label={ propertyMetadata?.displayName }
+                required={ propertyMetadata?.isMandatory }
+                requiredErrorMessage={ I18n.instance.t("console:develop.features.authenticationProvider." +
+                    "forms.common.requiredErrorMessage") }
+                type="dropdown"
+                value={ eachProp?.value }
+                key={ eachProp?.key || propertyMetadata?.key }
+                children={ getDropDownChildren(eachProp?.key, propertyMetadata?.options) }
+                disabled={ propertyMetadata?.isDisabled }
+                readOnly={ propertyMetadata?.readOnly }
+                data-testid={ `${ testId }-${ propertyMetadata?.key }` }
+                listen={ (values: Map<string, FormValue>) => {
+                    listen(propertyMetadata?.key, values);
+                } }
+                default ={ propertyMetadata?.defaultValue }
+            />) }
+            {
+                showField
+                && propertyMetadata?.description
+                && (
+                    <Hint disabled={ propertyMetadata?.isDisabled }>
+                        { propertyMetadata?.description }
+                    </Hint>
+                )
+            }
+        </>
+    );
+};
+
+export const getFilePicker = (eachProp: CommonPluggableComponentPropertyInterface,
```

Comment:
Shall we use a better variable name here instead of `eachProp` if possible?

Reply:
This variable is used in many places of the file. Hence keeping it to maintain consistency.